src/controllers/plugins/LuaUtilities.hpp

```diff
@@ -150,6 +254,7 @@ StackIdx push(lua_State *L, T inp)
 template <typename T>
 bool pop(lua_State *L, T *out, StackIdx idx = -1)
 {
+    BalanceKepper _(L, -1);
```

Comment:
Let's give this & uses like it names in the future - for mutex guards.
For mutex guards, we'd still do something like
```c++
std::unique_lock lock(this->mutex);
```